mis/data/graphs.py

```diff
@@ -538,3 +540,39 @@ def ingest(self, graph: NXWithPos, device: pl.devices.Device, id: int) -> BaseGr
         pyg.pos = torch.tensor(positions, dtype=torch.float)
 
         return BaseGraph(id=id, device=device, data=pyg)
+
+
+@dataclass
+class DIMACSDataset:
+    instance: MISInstance
+    solutions: list[list[int]]
+
+
+def load_dimacs(path: str) -> DIMACSDataset:
+    """
+    Load a DIMACS file and return a DIMACSDataset.
+
+    Args:
+        path (str): Path to the DIMACS file.
+
+    Returns:
+        DIMACSDataset: An instance of DIMACSDataset.
+    """
+    with open(path, 'r') as f:
+        lines = f.readlines()
+
+    # Parse the graph from the DIMACS format
+    edges = []
+    for line in lines:
+        if line.startswith('p'):
+            continue  # Skip problem line
```

Comment:
@RolandMacDoland Do you think it would be useful to actually load this line and test that we have the right number of edges ad nodes?

Reply:
Yes, a more elaborated version can be found [here](https://github.com/pasqal-io/maximum-independent-set/pull/28/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128)

Reply:
Ok done